learning_journal/views/default.py

```diff
@@ -1,20 +1,93 @@
 from pyramid.response import Response
 from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound
+from pyramid.httpexceptions import exception_response
+from pyramid.security import remember, forget
+from learning_journal.security import check_credentials
+
+
+import time
+import datetime
 
 from sqlalchemy.exc import DBAPIError
 
-from ..models import MyModel
+from ..models import Entry
+
+
+@view_config(route_name='login', renderer='../templates/login.jinja2', permission="view")
+def login_view(request):
+    """Handle the login view."""
+    if request.POST:
+        username = request.POST["username"]
+        password = request.POST["password"]
+        if check_credentials(username, password):
+            auth_head = remember(request, username)
+            return HTTPFound(
+                location=request.route_url("homepage"),
+                headers=auth_head
+            )
+
+    return {}
 
 
-@view_config(route_name='home', renderer='../templates/mytemplate.jinja2')
+@view_config(route_name='logout')
+def logout_view(request):
+    """Handle logging the user out."""
+    headers = forget(request)
+    return HTTPFound(request.route_url("homepage"), headers=headers)
+
+
+@view_config(route_name='homepage', renderer='../templates/index.jinja2', permission="view")
 def my_view(request):
```

Comment:
Why is this still called `my_view`? If it's the view for the home page then make its name match its function.